### PR TITLE
add custom scalar Specification URL

### PR DESCRIPTION
**Pull request by andimarek**
This is a major new feature for GraphQL: it allows you to specify a url for a custom scalar which should point to a document describing this scalar.

It is implemented via directive for SDL or as property on GraphQLScalar.

```graphql
scalar MyScalar @specifiedBy(url: "https://example.com/myScala")
...
```
The specifiedByUrl can then be queried via Introspection: 
```graphql
{__type(name: "MyScalar") {name specifiedByUrl} }
```

See https://github.com/graphql/graphql-spec/pull/649 and https://github.com/graphql/graphql-js/pull/2276 for all the details.

### CODE REVIEW

File: src/main/java/graphql/Directives.java [hunk 1]
```diff
@@ -64,11 +65,23 @@
...
                     .name("reason")
-                    .type(nonNull(GraphQLString))
+                    .type(GraphQLString)
                     .defaultValue("No longer supported")
                     .description("The reason for the deprecation"))
             .validLocations(FIELD_DEFINITION, ENUM_VALUE)
             .build();
 
+    /**
+     * The "specifiedBy" directive allows to provide a specification URL for a Scala
```
Comment (bbakerman) [inline src/main/java/graphql/Directives.java:75]: spillin - Scalar (not Scala - never Scala)

File: src/main/java/graphql/schema/GraphQLScalarType.java [hunk 2]
```diff
@@ -73,10 +74,16 @@ public GraphQLScalarType(String name, String description, Coercing coercing) {
     @Internal
     @Deprecated
     public GraphQLScalarType(String name, String description, Coercing coercing, List<GraphQLDirective> directives, ScalarTypeDefinition definition) {
```
Comment (bbakerman) [inline src/main/java/graphql/schema/GraphQLScalarType.java:76]: one version we need to remove the old constructors!

File: src/main/java/graphql/schema/idl/SchemaGenerator.java [hunk 3]
```diff
@@ -694,17 +697,32 @@ private GraphQLScalarType buildScalar(BuildContext buildCtx, ScalarTypeDefinitio
...
+        List<Directive> allDirectives = new ArrayList<>(scalarTypeDefinition.getDirectives());
+        extensions.forEach(extension -> allDirectives.addAll(extension.getDirectives()));
+        Optional<Directive> specifiedByDirective = FpKit.findOne(allDirectives,
+                directiveDefinition -> directiveDefinition.getName().equals(Directives.SpecifiedByDirective.getName()));
+        if (!specifiedByDirective.isPresent()) {
+            return null;
+        }
+        Argument urlArgument = specifiedByDirective.get().getArgument("url");
+        StringValue url = (StringValue) urlArgument.getValue();
+        return url.getValue();
```
Comment (bbakerman) [inline src/main/java/graphql/schema/idl/SchemaGenerator.java:723]: Thought - Will it always be non null arg values?

Comment (andimarek) [reply src/main/java/graphql/schema/idl/SchemaGenerator.java:723]: yes it is: url is required non null

**Pull request merged**